fallback/s2a_fallback.go

```diff
@@ -0,0 +1,121 @@
+package fallback
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/grpclog"
+	"net"
+)
+
+const (
+	alpnProtoStrH2   = "h2"
+	defaultHttpsPort = "443"
+)
+
+// DefaultFallbackClientHandshakeFunc returns an implementation of FallbackOptions.FallbackClientHandshakeFunc.
+// example use:
+//
+//	transportCreds, _ = s2a.NewClientCreds(&s2a.ClientOptions{
+//		S2AAddress: s2aAddress,
+//		EnableV2:   true,
+//		FallbackOpts: &s2a.FallbackOptions{ // optional, and for V2 only
+//			FallbackClientHandshakeFunc: fallback.DefaultFallbackClientHandshakeFunc(fallbackAddr),
+//		},
+//	})
+//
+// The fallbackAddr is expected to be a network address, e.g. example.com:port. If port is not specified,
+// it uses default port 443
+func DefaultFallbackClientHandshakeFunc(fallbackAddr string) func(context.Context, string, net.Conn, error) (net.Conn, credentials.AuthInfo, error) {
+	return func(ctx context.Context, originServer string, originConn net.Conn, originErr error) (net.Conn, credentials.AuthInfo, error) {
```

Comment:
Can you clarify the meaning of the originServer and originConn and originErr parameters?

It's unclear to me what these mean.